docs/topics/upgrade.md

```diff
@@ -18,7 +18,15 @@ In order to ensure that your `aks-engine upgrade` operation runs smoothly, there
 
 1) You will need access to the original `_output/` directory files that were generated during cluster deployment in order to run `upgrade`.  `aks-engine` will use the `--deployment-dir` argument provided in the upgrade command to introspect the `apimodel.json` file and Azure Resource Manager (ARM) parameters in order to determine the cluster's current Kubernetes version, as well as all other cluster configuration data as defined by `aks-engine` during the last time that `aks-engine` was used to deploy, scale, or upgrade the cluster.
 
-2) `aks-engine upgrade` expects a cluster configuration that conforms to the current state of the cluster. In other words, the Azure resources inside the resource group deployed by `aks-engine` should be in the same state that when they were originally created by `aks-engine`. If you perform manual operations on your Azure IaaS resources (other than `aks-engine scale` and `aks-engine upgrade`) DO NOT use `aks-engine upgrade`, as the aks-engine-generated ARM template won't be reconcilable against the state of the Azure resources that reside in the resource group. This includes naming of resources; `aks-engine` relies on some resources (such as VMs) to be named in accordance with the original `aks-engine` deployment. In summary, the set of Azure resources in the resource group are mutually reconcilable only if they have been exclusively created and managed as the result of a series of successive ARM template deployments originating from `aks-engine`.
+2) `aks-engine upgrade` expects a cluster configuration that conforms to the current state of the cluster. In other words, the Azure resources inside the resource group deployed by `aks-engine` should be in the same state that when they were originally created by `aks-engine`. If you do any manual changes against the resources in the resource group created by `aks-engine` DO NOT use `aks-engine upgrade`, as the aks-engine-generated ARM template won't be reconcilable against the state of the Azure resources that reside in that resource group. This includes:
+
+- Any manual changes to resources via the `az` CLI
+- Any manual changes to resources via the Azure Portal
+- Any manual changes to resources via an ARM template deployment against the `aks-engine`-created resource group
+- Any manual changes to resources via Azure Resource Manager (https://resources.azure.com/)
+- Any manual changes via a 3rd party tool (e.g., ansible, terraform, chef) that employ the above resource modification vectors
```

Comment:
I'm not sure listing specific examples is a good idea because people might ignore the "includes" part and assume that if something they are doing is not on this list it's fine.

Also the sorter and more concise we keep these recommendations, the more likely users are to actually read it and it's already pretty long. My eyes personally tend to jump to bullet points when I look at a big doc and ignore big paragraphs...